pageserver/src/layered_repository/disk_btree.rs

```diff
@@ -0,0 +1,3012 @@
+        let expected = all_data
+            .iter()
+            .rev()
+            .map(|(&key, &val)| (key, val))
+            .collect::<Vec<(u64, u64)>>();
+        assert_eq!(*result.lock().unwrap(), expected);
+
+        Ok(())
+    }
+
+    #[test]
+    fn random_data() -> anyhow::Result<()> {
+        // Generate random keys with exponential distribution, to
+        // exercise the prefix compression
+        const NUM_KEYS: usize = 100000;
+        let mut all_data: BTreeMap<u128, u64> = BTreeMap::new();
+        for idx in 0..NUM_KEYS {
+            let u: f64 = rand::thread_rng().gen_range(0.0..1.0);
+            let t = -(f64::ln(u));
+            let key_int = (t * 1000000.0) as u128;
+
+            all_data.insert(key_int as u128, idx as u64);
+        }
+
+        // Build a tree from it
+        let mut disk = TestDisk::new();
+        let mut writer = DiskBtreeBuilder::new(16, &mut disk);
+
+        for (&key, &val) in all_data.iter() {
+            writer.append(&u128::to_be_bytes(key), val)?;
+        }
+        let (root_offset, _writer) = writer.finish()?;
+
+        let reader = DiskBtreeReader::new(root_offset, disk);
+
+        // Test get() operation on all the keys
+        for (&key, &val) in all_data.iter() {
+            let search_key = u128::to_be_bytes(key);
+            assert_eq!(reader.get(&search_key)?, Some(val));
+        }
+
+        // Test get() operations on random keys, most of which will not exist
+        for _ in 0..100000 {
+            let key_int = rand::thread_rng().gen::<u128>();
+            let search_key = u128::to_be_bytes(key_int);
+            assert!(reader.get(&search_key)? == all_data.get(&key_int).cloned());
+        }
+
+        // Test boundary cases
+        assert!(reader.get(&u128::to_be_bytes(u128::MIN))? == all_data.get(&u128::MIN).cloned());
+        assert!(reader.get(&u128::to_be_bytes(u128::MAX))? == all_data.get(&u128::MAX).cloned());
+
+        Ok(())
+    }
+
+    #[test]
+    #[should_panic(expected = "unsorted input")]
+    fn unsorted_input() {
+        let mut disk = TestDisk::new();
+        let mut writer = DiskBtreeBuilder::new(2, &mut disk);
+
+        let _ = writer.append(b"ba", 1);
+        let _ = writer.append(b"bb", 2);
+        let _ = writer.append(b"aa", 3);
+    }
+
+    ///
+    /// This test contains a particular data set, representing all the keys
+    /// generated by the 'test_random_updates' unit test. I extracted this while
+    /// trying to debug a failure in that test. The bug turned out to be
+    /// elsewhere, and I'm not sure if this is still useful, but keeping it for
+    /// now...  Maybe it's a useful data set to show the typical key-values used
+    /// by a delta layer, for evaluating how well the prefix compression works.
+    ///
+    #[test]
+    fn particular_data() -> anyhow::Result<()> {
+        #[rustfmt::skip]
+        static DATA:  [([u8; 26], u64); 2000] = [
```

Comment:
If we keep this test data set I suggest to split it out to a separate file to increase readability. There is a handy macro [include_str!](https://doc.rust-lang.org/std/macro.include_str.html) that can help with that. Included string can have rows with two sort of columns separated by a space for example.

Reply:
I moved it to a separate .rs file, without `include_str!`.